ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -95,6 +127,38 @@ message Commands {
   // to read contract data on behalf of each of the given parties.
   // Optional
   repeated string read_as = 13;
+
+  // A unique identifier to distinguish completions for different submissions with the same change ID.
+  // Typically a random UUID. Applications are expected to use a different UUID for each retry of a submission
+  // with the same change ID.
+  //
+  // If omitted, the participant or Daml ledger may set a UUID of their choice.
+  // Optional
+  //
+  // TODO(v2) Do we want to enforce the format of a UUID here?
```

Comment:
No, but add the constraint that it is a LedgerString

Reply:
Changed.